server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -86,15 +89,46 @@ public final void cleanupNodeSettings() {
         );
     }
 
+    public void testSearchTaskCancellationWithHighElapsedTime() throws InterruptedException {
+        Settings request = Settings.builder()
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(SearchTaskSettings.SETTING_ELAPSED_TIME_MILLIS_THRESHOLD.getKey(), 1000)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        ExceptionCatchingListener listener = new ExceptionCatchingListener();
+        client().execute(
+            TestTransportAction.ACTION,
+            new TestRequest<>(
+                RequestType.HIGH_ELAPSED_TIME,
+                (TaskFactory<Task>) (id, type, action, description, parentTaskId, headers) -> new SearchTask(
+                    id,
+                    type,
+                    action,
+                    descriptionSupplier(description),
+                    parentTaskId,
+                    headers
+                )
+            ),
+            listener
+        );
+        assertTrue(listener.latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS));
+
+        Exception caughtException = listener.getException();
+        assertNotNull("SearchTask should have been cancelled with TaskCancelledException", caughtException);
+        MatcherAssert.assertThat(caughtException, instanceOf(TaskCancelledException.class));
+        MatcherAssert.assertThat(caughtException.getMessage(), containsString("elapsed time exceeded"));
```

Comment:
Do we also need to verify if the cancellation count in the stats API has got incremented? Similarly for all tests.

Reply:
I will add that in the upcoming PRs.